libs/gl-client/src/lnurl/tor_http_client.rs

```diff
@@ -0,0 +1,48 @@
+use anyhow::Result;
+use arti_client::{TorClient, TorClientConfig};
+use log::info;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use url::Url;
+
+pub struct TorHttpClient {
+    tor_client: TorClient<tor_rtcompat::PreferredRuntime>,
+}
+
+impl TorHttpClient {
+    pub async fn new() -> Result<Self> {
+        let mut config_builder = TorClientConfig::builder();
+        let address_filter = config_builder.address_filter();
+        address_filter.allow_onion_addrs(true);
+        address_filter.build()?;
+        config_builder.build()?;
+
+        let tor_client = TorClient::create_bootstrapped(config_builder.build().unwrap()).await?;
```

Comment:
```suggestion
        let tor_client = TorClient::create_bootstrapped(config_builder.build()?).await?;
```

Reply:
Notice too that `create_bootstrapped` is actually going out and establishing Tor circuits, which can take considerable time. If we then end up not using that circuit all of that work is wasted. It'd likely be a better idea to keep the builder around, and defer bootstrapping to when we know we need the circuit.